### PR TITLE
NODE-127, fix: add psbt fee verification logic

### PR DESCRIPTION
**Pull request by alstjd0921**
## Description

* add psbt fee verification logic

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Something else (simple changes that are not related to existing functionality or others)

# Checklist

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have made new test codes regarding to my changes.
- [ ] I have no personal secrets or credentials described on my changes.
- [ ] I have run `cargo-clippy`  and linted my code.
- [ ] My changes generate no new warnings.
- [ ] My changes passed the existing test codes.
- [ ] My changes are able to compile.

### CODE REVIEW

File: pallets/btc-registration-pool/Cargo.toml [hunk 1]
```diff
@@ -28,12 +28,13 @@ bp-multi-sig = { workspace = true }
...
+    "scale-info/std",
+    "parity-scale-codec/std",
+    "frame-support/std",
+    "frame-system/std",
+    "sp-runtime/std",
+    "sp-std/std",
+    "sp-core/std",
+    "bp-multi-sig/std",
 ]
+try-runtime = ["frame-support/try-runtime"]
```
Comment (dnjscksdn98) [inline pallets/btc-registration-pool/Cargo.toml:40]: 이거 왜 추가된거에요?

File: pallets/btc-socket-queue/src/pallet/mod.rs [hunk 2]
```diff
@@ -608,6 +620,26 @@ pub mod pallet {
...
+			ensure_root(origin)?;
+
+			let old = <MaxFeeRate<T>>::get();
+			ensure!(old != new, Error::<T>::NoWritingSameValue);
+
+			// overflow check
+			match FeeRate::from_sat_per_vb(new) {
+				Some(_) => {},
+				None => return Err(Error::<T>::OutOfRange.into()),
+			};
```
Comment (dnjscksdn98) [inline pallets/btc-socket-queue/src/pallet/mod.rs:636]:
```suggestion
			FeeRate::from_sat_per_vb(new).ok_or(Error::<T>::OutOfRange)?;
```
한줄로 간략하게도 가능합니다

File: pallets/btc-socket-queue/src/pallet/mod.rs [hunk 3]
```diff
@@ -207,6 +212,10 @@ pub mod pallet {
 	pub type BondedRollbackOutputs<T: Config> =
 		StorageDoubleMap<_, Twox64Concat, H256, Twox64Concat, U256, H256>;
 
+	#[pallet::storage]
+	/// The maximum fee rate(sat/vb) that can be set for PSBT.
+	pub type MaxFeeRate<T: Config> = StorageValue<_, u64, ValueQuery>;
```
Comment (dnjscksdn98) [inline pallets/btc-socket-queue/src/pallet/mod.rs:217]: Config trait에 디폴트 값 넣을수 있도록 하는것도 필요할것 같아요

File: pallets/btc-socket-queue/src/pallet/mod.rs [hunk 4]
```diff
@@ -311,6 +320,9 @@ pub mod pallet {
 			let psbt_obj = Self::try_get_checked_psbt(&psbt)?;
 			let txid = Self::convert_txid(psbt_obj.unsigned_tx.txid());
 
+			// verify if the fee rate is set properly
+			Self::try_psbt_fee_verification(&psbt_obj)?;
```
Comment (dnjscksdn98) [inline pallets/btc-socket-queue/src/pallet/mod.rs:324]: `submit_migration_request`, `submit_rollback_request`에도 추가해야 될것 같아요

File: pallets/btc-socket-queue/src/pallet/mod.rs (hunk 3 above)
Comment (dnjscksdn98) [reply pallets/btc-socket-queue/src/pallet/mod.rs:217]: 그리고 psbt-manager에서도 라이브러리에서 제공되는 rate 말고, 노드에 저장된 값 불러와서 사용해야 될것 같네요

Comment (dnjscksdn98) [reply pallets/btc-socket-queue/src/pallet/mod.rs:217]: 아니면 config에 빼도 되구요

File: pallets/btc-registration-pool/Cargo.toml (hunk 1 above)
Comment (alstjd0921) [reply pallets/btc-registration-pool/Cargo.toml:40]: cargo warning 보이길래 넣었습니다 (unexpected cfg condition value)

File: pallets/btc-socket-queue/src/pallet/mod.rs (hunk 4 above)
Comment (alstjd0921) [reply pallets/btc-socket-queue/src/pallet/mod.rs:324]: https://github.com/bifrost-platform/bifrost-node/pull/69/commits/b750af8b4b7f972684b4fde20fa023e2912b6eb3

**Review by dnjscksdn98 — APPROVED** [review]

**Pull request merged**